lib/utils/requireModule.js

```diff
@@ -9,40 +9,55 @@ module.exports = function (fullpath) {
   try {
     exported = require(fullpath);
   } catch (err) {
-    if (err.message === 'Cannot use import statement outside a module' || err.message === 'Unexpected token \'export\'') {
-      err.detailedErr = err.message;
-      err.help = ['Using ES6 import/export syntax? - make sure to specify "type=module" in your package.json or use .mjs extension.'];
-      err.link = 'https://nodejs.org/api/esm.html';
+    const isEsmSyntaxError = err.message === 'Cannot use import statement outside a module' ||
+      err.message.includes('Unexpected token \'export\'');
+    const isMjsFile = fullpath.endsWith('.mjs');
 
-      throw err;
-    }
+    if (err.code === 'ERR_REQUIRE_ESM' || (isMjsFile && isEsmSyntaxError)) {
```

Comment:
add a comment that in case `ts-node` is activated, `require()` intercepted by `ts-node` could throw ESM syntax error for Node 20+ even for `.mjs` file (for Node 18 and below, `ERR_REQUIRE_ESM` is thrown before the call is intercepted by `ts-node`).